backend/src/graphql/resolver/UserResolver.ts

```diff
@@ -788,19 +873,58 @@ export class UserResolver {
   }
 }
 
+async function findUserByEmail(email: string): Promise<DbUser> {
+  const dbUserContact = await DbUserContact.findOneOrFail(
+    { email: email },
+    { withDeleted: true, relations: ['user'] },
+  ).catch(() => {
+    logger.error(`UserContact with email=${email} does not exists`)
+    throw new Error('No user with this credentials')
+  })
+  const dbUser = dbUserContact.user
+  dbUser.emailContact = dbUserContact
+  return dbUser
+}
+
+async function checkEmailExists(email: string): Promise<boolean> {
+  const userContact = await DbUserContact.findOne({ email: email }, { withDeleted: true })
+  if (userContact) {
+    return true
+  }
+  return false
+}
+
+/*
 const isTimeExpired = (optIn: LoginEmailOptIn, duration: number): boolean => {
   const timeElapsed = Date.now() - new Date(optIn.updatedAt).getTime()
   // time is given in minutes
   return timeElapsed <= duration * 60 * 1000
 }
-
+*/
+const isTimeExpired = (updatedAt: Date, duration: number): boolean => {
+  const timeElapsed = Date.now() - new Date(updatedAt).getTime()
+  // time is given in minutes
+  return timeElapsed <= duration * 60 * 1000
+}
+/*
 const isOptInValid = (optIn: LoginEmailOptIn): boolean => {
   return isTimeExpired(optIn, CONFIG.EMAIL_CODE_VALID_TIME)
 }
-
+*/
+const isEmailVerificationCodeValid = (updatedAt: Date | null): boolean => {
+  if (updatedAt == null) {
```

Comment:
It seems a missunderstanding here.
it checks if the udaptedAt is not set yet, that means it is the original code from the first creation.
If the updatedAt exists, then is is necessary to validate the duration since the code was updated.